sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -76,7 +81,23 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
           createdAt: currentDateTime,
         } as StudentAppealRequest),
     );
-    return this.repo.save(studentAppeal);
+
+    // Update the student appeal and files if exist using the same entity manager
+    // so that they are processed within the same transaction
+    return await this.dataSource.transaction(async (entityManager) => {
```

Comment:
this.dataSource.transaction can be returned which will be awaited at consumer end.